File: frontend/src/models/link.ts
import {ID} from "./base";

export type Link = {
    id: ID;
    url: string;
    name: string;
    description?: string;
}

export type NewLink = {
    url: string;
    name: string;
    description?: string;
}

export type UpdateLink = {
    pageId?: ID;
    url?: string;
    name?: string;
    description?: string;
};

File: ragbag-webapp/src/main/java/de/serbroda/ragbag/models/Link.java
package de.serbroda.ragbag.models;

import de.serbroda.ragbag.models.base.AbstractBaseEntity;
import jakarta.persistence.*;

@Entity
@Table(name = "link")
public class Link extends AbstractBaseEntity {

    private String name;
    private String url;
    private Page page;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Override
    public Long getId() {
        return doGetId();
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    @ManyToOne(fetch = FetchType.EAGER)
    public Page getPage() {
        return page;
    }

    public void setPage(Page page) {
        this.page = page;
    }
}
